rust/connlib/tunnel/src/gateway.rs

```diff
@@ -382,7 +401,17 @@ impl GatewayState {
     }
 
     pub(crate) fn poll_event(&mut self) -> Option<GatewayEvent> {
-        self.buffered_events.pop_front()
+        if let Some(ev) = self.buffered_events.pop_front() {
+            return Some(ev);
+        }
+
+        for peer in self.peers.iter_mut() {
+            if let Some(ev) = peer.poll_event() {
+                return Some(ev);
+            }
+        }
```

Comment:
In `snownet`, I am doing this kind of work in `handle_timeout` and just move all the events in `buffered_events`, you can do something like:

```rust
self.buffered_events.extend(self.peers.iter_mut().flat_map(|peer| iter::from_fn(|| peer.poll_event())));
```

Reply:
It is better to treat `handle_timeout` as the kind of "work-horse" of these state machines and keep getters like `poll_event` simple.

Reply:
so you move the `buffered_events` in the state `handle_timeout`?